### DISCUSSION TITLE
How to add Curator picker to markdown editor

### DISCUSSION THREAD

**Discussion by GasparSukk** [awcodes/filament-curator#441] [Q&A]
I would like to add/edit/remove images in my markdown editor using Curator plugin/picker.

Is it possible? How can i do it?

Screenshot attached.
<img width="564" alt="image" src="https://github.com/awcodes/filament-curator/assets/95946/53ec9a56-c237-455c-b063-8fb80a1503c7">

**Comment by awcodes** [comment]
That's not possible without updates to the core markdown editor.

**Comment by GasparSukk** [comment]
What about extending Markdown editor?
https://github.com/spatie/filament-markdown-editor
```php
use Spatie\FilamentMarkdownEditor\MarkdownEditor;

public static function form(Form $form): Form
{
    return $form
        ->schema([
            // other fields ...

            MarkdownEditor::make('text')
                ->fileAttachmentsDisk('admin-uploads')
                ->fileAttachmentsVisibility('public')
                ->required(),
        ]);
}

**Reply by awcodes** [reply]
That's still not going to work with Curator, because Curator dispatches an event with the selected files. So the editor has to have a listener for the event so it can know what to do with the data.

**Comment by GasparSukk** [comment]
How to resolve the issue with content with images?
Does your Tiptap Editor supports your Curator plugin?

**Reply by awcodes** [reply]
Yes Tiptap Editor works with Curator. It's a setting in the config.
